feat(dart/transform): Add simple ParseTemplates step

kegluneq opened the pull request:
Add a step + tests for parseing `inline` Template values to generate getters, setters, and methods.

Commit:
feat(dart/transform): Implement `Html5LibDomAdapter` methods.

Adding support for some additional `DomAdapter` methods necessary to run
`Complier` in the Dart transformer.

Commit:
feat(dart/transform): Add simple ParseTemplates step

Adds a step that parses `inline` Template values to generate getters and
setters.

Commit:
feat(dart/transform): Add simple ParseTemplates step

Generate methods in the ParseTemplates step.
Add a test for inline template method generation.

Commit:
fix(dart/transform): Save & restore `ReflectionCapabilities`

Restore `ReflectionCapabilities` once finished with the `ParseTemplates`
transform step.

yjbanov commented on modules/angular2/src/transform/template_parser/generator.dart:
```diff
@@ -0,0 +1,141 @@
+
+Iterable<String> _generateSetters(String typeName, List<String> setterName) {
+  return setterName.map((prop) {
+    return ''''$prop': (o, String v) => o.$prop = v''';
```
`v` always a `String`?

kegluneq replied:
Maybe my understanding of how this will be used is not clear. Can you provide an example of when this value would not be a `String`?

Commit:
style(dart/transform): Rename TemplateParser > TemplateCompiler

More accurately reflect what the step is doing.

kegluneq commented:
@yjbanov think I addressed all your comments
- Added https://github.com/angular/angular/issues/1005 to remove `src` from library directives
- Renamed `TemplateParser` > `TemplateCompiler`

The only thing I still see is the `String` type annotation for the generated setters. In what situation would the passed in value not be a `String`?

yjbanov commented:
The value is bound from the application and it could be anything, String, num, bool, whatever.

kegluneq commented:
Okay, removed the `String` annotation, think this is all set to go in (once Travis is happy)

Commit:
style(dart/transform): Rename .ngDeps.dart => .ng_deps.dart

Update to conform with file naming conventions.
See https://github.com/angular/angular/pull/975

kegluneq commented:
Closed by 08b56e1c532d23120cdcc86f2fd95de29271bc91
